streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -737,6 +754,61 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
             thread.setDaemon(true);
             return thread;
         });
+
+        // register instance-level metrics
+        final Map<String, String> clientTag = Collections.singletonMap(CLIENT_THREAD_ID_TAG, clientId);
+        AppInfoParser.registerAppInfo(JMX_PREFIX, clientId, metrics);
+        metrics.addMetric(
+            new MetricName(
+                "application-id",
```

Comment:
should `application-id`,  `state`,  and associated descriptions be declared as `static final` instance variables?

Reply:
I'm not sure... This is in the constructor for KafkaStreams, so the only duplication it would save is if people ran more than one Streams instance in the JVM. Seems like kind of a low ROI for adding even more fields (with just one usage) to the object.